### PR TITLE
Fix e2e-old-babel test to always use old core/helpers/traverse

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14794"></a>

### CODE REVIEW

**Comment by liuxingbaoyu** [comment]
image
`resolutions` doesn't seem to work.

**Comment by nicolo-ribaudo** [comment]
The gulpfile updates in https://github.com/babel/babel/pull/14793 should fix this!

**Comment by liuxingbaoyu** [comment]
That's great!
Are the current tests working as expected?
It looks like `@babel/traverse` which `@babel/core` depends on should also be pinned to `7.0.0`.
When I changed I got a new error, `ReferenceError: Unknown helper classStaticPrivateFieldSpecGet`, I don't know how to fix this kind of error.

**Comment by nicolo-ribaudo** [comment]
Yeah, we currently only pin _some_ packages to simulate "bad" user configurations, where they may have an old `@babel/core` version but its deps are free to change versions.

**Comment by liuxingbaoyu** [comment]
```shell
(
  # Yarn prints colors on GH actions even if it's piped, unless explicitly disabled
  # https://github.com/yarnpkg/berry/pull/659
  YARN_ENABLE_COLORS=0 yarn why @babel/core | grep -o "@babel/core@npm:.* (via npm:.*)";
  YARN_ENABLE_COLORS=0 yarn why @babel/helpers | grep -o "@babel/helpers@npm:.* (via npm:.*)";
  YARN_ENABLE_COLORS=0 yarn why @babel/traverse | grep -o "@babel/traverse@npm:.* (via npm:.*)"
) | uniq | node -e "
  var pkg = require('./package.json');
  var packages = fs.readFileSync(0, 'utf8').trim().split('\n');
  pkg.devDependencies['@babel/core'] = '7.0.0';
  packages.forEach(desc => {
    const { name, specifier } = desc
      .match(/(?<name>@babel\/[a-z]+).*\(via (?<specifier>npm:[^)]+)\)/)
      .groups;
    pkg.resolutions[name + '@' + specifier] = '7.0.0';
  });
  fs.writeFileSync('./package.json', JSON.stringify(pkg, null, 2));
"
```

But our current script also seems to be pinning both `@babel/helpers` and `@babel/traverse`.😕
Like the screenshot in https://github.com/babel/babel/pull/14794#issuecomment-1193359262, they were added to `resolutions`, just didn't work for some reason.

File: scripts/integration-tests/e2e-babel-old-version.sh [hunk 1]
```diff
@@ -24,26 +24,15 @@ startLocalRegistry "$PWD"/scripts/integration-tests/verdaccio-config.yml
...
-    pkg.resolutions[name + '@' + specifier] = '7.0.0';
-  });
-
-  fs.writeFileSync('./package.json', JSON.stringify(pkg, null, 2));
+node -e "\
+  var pkg = require('./package.json');\
+  pkg.devDependencies['@babel/core'] = '7.0.0';\
+  Object.assign(pkg.resolutions, {\
+    '@babel/core': '7.0.0',\
+    '@babel/helpers': '7.0.2',\
```
Comment (nicolo-ribaudo) [inline scripts/integration-tests/e2e-babel-old-version.sh:32]:
```suggestion
    '@babel/helpers': '7.1.0',\
```

**Pull request merged**